RFCs/rfc-0007-zkPassport.md

```diff
@@ -0,0 +1,106 @@
+- **Submitted by**: Evan Shapiro (Github: @es92, Twitter: @evanashapiro)
+- **Submitted on**: 1/8/2024
+
+## Abstract
+
+This RFC proposes a grant to build a private proof-of-passport attestation (zkPassport) leveraging smartphone NFC, ZKPs, and Mina, with an aim to provide a private identity primitive for Mina, blockchains that are connected to it, and ultimately web2 applications.
+
+## Introduction
+
+A secure, widely accessible implementation of private identity (proof-of-personhood) would be very useful for the blockchain ecosystem and for digital systems broadly.
+
+This could be leveraged by applications on Mina, applications on any bridged chains (eg Ethereum), or any web3-connected web2 applications.
+
+Use cases include:
+* Proving proof of uniqueness, to apply to use cases such as one person one vote, one person one account, one person one airdrop, etc.
+* Proving facts about identity (nationality, age, etc).
+
+Historically, this has been challenging to bootstrap and make accessible without some prior information to "anchor" digital identity to.
+
+One widely accessible place that identity information has been stored, including with cryptographic signatures and roots of trust, is passports. It seems that for decades now, passports have come with (1) NFC, (2) digital representations of their data, and (3) cryptographic signatures by their host countries[[1](https://www.icao.int/Security/FAL/PKD/Pages/ePassport-Basics.aspx),[2](https://en.wikipedia.org/wiki/Biometric_passport)]. As of writing, it seems there are currently over 140 countries issuing biometric passports.
+
+Using this information directly online would expose passport information online, which would not be viable.
+
+However, with ZKPs, users could provide proofs about their passport information, without exposing the passport information itself.
+
+Additionally, because passports following this standard support NFC, which is present on most smartphones, users can obtain this data themselves from their own passports relatively easily. Here for example is an open library that reads passports for iOS[[3](https://github.com/AndyQ/NFCPassportReader)].
+
+Properties stored in the standard include:
+* documentType
+* documentSubType
+* documentNumber
+* issuingAuthority
+* documentExpiryDate
+* dateOfBirth
+* gender
+* nationality
+* lastName
+* firstName
+* names
+* placeOfBirth
+* residenceAddress
+* phoneNumber
+
+## Objectives
+
+We propose a grant taking advantage of this, implementing, at a high level, the following:
+
+1. An iOS & Android App that uses NFC to allow users to get the data from their own passports, and store that data in their MINA wallet for secure use by web3 applications through the (in-progress) Mina Attestation API Standard.
+2. An o1js library that uses the Mina Attestation API Standard to create proofs of various components of one's zkPassport identity, for use in proof-of-unique-user and proof of nationality & age use cases.
```

Comment:
Very curious if there is a need for a iOS & Android App when NFC could be connected with a standard browser 

Tested the following website with my cheapish Android phone on Chrome

https://whatwebcando.today/nfc.html

Reply:
Using NFC scanning directly in the browser instead of creating separate iOS & Android app would make the process more modular and user-friendly, I think. It would be simpler and more direct: go to a website, scan your passport using NFC, and then transfer the data to your wallet, zkApp, etc.

I think the Web NFC API currently only supports basic scanning functions. With a passport NFC scan, there's an additional step to decrypt the data using a key found on the passport's first page. This particular interaction isn't included in the native Web NFC API, I think. 

I hope this is achievable. Such a fantastic use case.

Reply:
Yep, the Web NFC API doesn't support reading passport unfortunately.